apps/shared/avifjpeg.c

```diff
@@ -749,12 +752,18 @@ static avifBool avifJPEGExtractGainMapImageFromMpf(FILE * f,
     AVIF_CHECK(avifJPEGFindMpfSegmentOffset(f, &mpfSegmentOffset));
 
     for (uint32_t imageIdx = 0; imageIdx < numImages; ++imageIdx) {
+        if (UINT32_MAX - offset < 4) {
+            return AVIF_FALSE;
+        }
         offset += 4; // Skip "Individual Image Attribute"
         uint32_t imageSize;
         AVIF_CHECK(avifJPEGReadU32(segmentData, &imageSize, &offset, isBigEndian));
         uint32_t imageDataOffset;
         AVIF_CHECK(avifJPEGReadU32(segmentData, &imageDataOffset, &offset, isBigEndian));
 
+        if (UINT32_MAX - offset < 4) {
+            return AVIF_FALSE;
+        }
```

Comment:
Maryla: I found these by searching for "offset +" in this file.

We also add numbers to `offset` in `avifJPEGFindMpfSegmentOffset()`. I did not add similar checks to that function because I think `offset` cannot possibly overflow in that function. It would be good if you could also audit `avifJPEGFindMpfSegmentOffset()` to verify my claim.

Reply:
Yes I agree with you. Thanks for the fix!